vnext/Scripts/IntegrationTests.ps1

```diff
@@ -1,97 +1,94 @@
+. $PSScriptRoot\TestServers.ps1
 
-$procs = @()
+if ($List) {
+	# https://github.com/Microsoft/vstest/issues/1732: Can't print full test names directly to console.
+	& $VsTest ($Assemblies -join ' ') --ListFullyQualifiedTests `
+		--ListTestsTargetPath:$env:TEMP\ReactWindowsIntegrationTestsList.txt
 
-if($List.IsPresent) {
-	Write-Host '-List is present. Skipping setup.'
-	$NoSetup = $true
+	Get-Content $env:TEMP\ReactWindowsIntegrationTestsList.txt
+	Exit
 }
 
-if (! $NoSetup) {
-	$procs += Start-Process powershell -PassThru -WorkingDirectory "$PSScriptRoot\.." -ArgumentList "npm run start"
-	$procs += Start-Process powershell -PassThru -ArgumentList '-NoProfile', "-File $ReactNativeHome\IntegrationTests\Launch-WebSocketServer.ps1 80 24"
+# Ensure test services are online.
+if (! $NoServers) {
+	$packager = Find-Packager
+	$wsServer = Find-WebSocketServer
+	$notFound = $false
+
+	if (!$packager) {
+		Write-Warning 'Packager not found. Attempting to start...'
+		Start-Packager -ReactNativeLocation ($PSScriptRoot | Split-Path) -NpmPath $NpmPath
+		$notFound = $true
+	} else {
+		Write-Host 'Found Packager.'
+	}
 
-	if ($Preload) {
+	if (!$wsServer) {
+		Write-Warning 'WebSocket server not found. Attempting to start...'
+		Start-WebSocketServer -ReactNativeLocation $ReactNativeLocation -NodePath $NodePath
+		$notFound = $true
+	} else {
+		Write-Host 'Found WebSocket server.'
+	}
+
+	if ($Preload -and $notFound) {
 		Start-Sleep -Seconds $Delay
 
 		# Preload the RNTesterApp integration bundle for better performance in integration tests.
 		Invoke-WebRequest -Uri "http://localhost:8081/IntegrationTests/IntegrationTestsAppWin.bundle?platform=windesktop&dev=true" | Out-Null
 	}
 }
 
-if ($Run) {
-	# Validate other options.
-	if ($List.IsPresent) {
-		Write-Warning '-Run switch is present. -List will be ignored.'
-	}
-	# Ensure server ports are open.
-	# 5555 - String-suffixing WebSocket server
-	# 8081 - Packager
-	$ports = (5555, 8081)
-	foreach ($port in $ports) {
-		try {
-			$tcpClient = New-Object System.Net.Sockets.TcpClient
-			$tcpClient.Connect("127.0.0.1", $port)
-			Write-Host "Found service on TCP port $port."
-		}
-		catch {
-			Write-Error "Test service on port $currentPort not found."
-			Exit
-		}
-		finally {
-			$tcpClient.Dispose()
-		}
-	}
-	# Run Integration Test assemblies.
-	& $VsTest $Assembly --InIsolation --Platform:$Platform ('', "--Tests:$($Tests -join ',')")[$Tests.Count -gt 0]
-} elseif ($List) {
-	#TODO: Make -NoSetup implied.
-	# https://github.com/Microsoft/vstest/issues/1732: Can't print full test names directly to console.
-	& $VsTest $Assembly --ListFullyQualifiedTests --ListTestsTargetPath:$env:TEMP\ReactWindowsIntegrationTestsList.txt
-	Get-Content $env:TEMP\ReactWindowsIntegrationTestsList.txt
+if ($NoRun) {
+	Exit
 }
 
-if ($Cleanup) {
-	Cleanup
-}
+# Run Integration Test assemblies.
+& $VsTest ($Assemblies -join ' ') --InIsolation --Platform:$Platform ('', "--Tests:$($Tests -join ',')")[$Tests.Count -gt 0]
```

Comment:
>('', "--Tests:$($Tests -join ',')")[$Tests.Count -gt 0] [](start = 69, length = 55)

He! The ternary operator that PS lacks?

Reply:
Yup, their ternary works as `($negativeResult, $positiveResult)[$condition]`.